hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/IncrementalContainerReportHandler.java

```diff
@@ -88,8 +88,10 @@ public void onMessage(final IncrementalContainerReportFromDatanode report,
             // Ensure we reuse the same ContainerID instance in containerInfo
             id = container.containerID();
           } finally {
-            if (!replicaProto.getState().equals(
+            if (replicaProto.getState().equals(
                 ContainerReplicaProto.State.DELETED)) {
+              nodeManager.removeContainer(dd, id);
+            } else {
```

Comment:
This looks like a GitHub PR diff display bug, which makes it look like it is brought in by the merge commit https://github.com/apache/ozone/pull/4134/commits/e7325c3b6d737d90f4dd647bcfd2ce567a1cab43

while the `master` branch already has the change: https://github.com/apache/ozone/blob/master/hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/IncrementalContainerReportHandler.java#L90-L97

It should be safe to ignore this.

Reply:
@smengcl Fixed it!